src/main/java/org/opentripplanner/model/impl/OtpTransitBuilder.java

```diff
@@ -0,0 +1,161 @@
+
+    private final List<Route> routes = new ArrayList<>();
+
+    private final List<ShapePoint> shapePoints = new ArrayList<>();
+
+    private final List<Stop> stops = new ArrayList<>();
+
+    private final List<StopTime> stopTimes = new ArrayList<>();
+
+    private final List<Transfer> transfers = new ArrayList<>();
+
+    private final List<Trip> trips = new ArrayList<>();
+
+    public OtpTransitBuilder() {
+    }
+
+    public OtpTransitBuilder(OtpTransitService transitService) {
+        add(transitService);
+    }
+
+    public OtpTransitBuilder add(OtpTransitService other) {
+        agencies.addAll(other.getAllAgencies());
+        calendarDates.addAll(other.getAllCalendarDates());
+        calendars.addAll(other.getAllCalendars());
+        fareAttributes.addAll(other.getAllFareAttributes());
+        fareRules.addAll(other.getAllFareRules());
+        feedInfos.addAll(other.getAllFeedInfos());
+        frequencies.addAll(other.getAllFrequencies());
+        pathways.addAll(other.getAllPathways());
+        routes.addAll(other.getAllRoutes());
+        shapePoints.addAll(other.getAllShapePoints());
+        stops.addAll(other.getAllStops());
+        stopTimes.addAll(other.getAllStopTimes());
+        transfers.addAll(other.getAllTransfers());
+        trips.addAll(other.getAllTrips());
+        return this;
+    }
+
+    public List<Agency> getAgencies() {
+        return agencies;
+    }
+
+    public List<ServiceCalendarDate> getCalendarDates() {
+        return calendarDates;
+    }
+
+    public List<ServiceCalendar> getCalendars() {
+        return calendars;
+    }
+
+    public List<FareAttribute> getFareAttributes() {
+        return fareAttributes;
+    }
+
+    public List<FareRule> getFareRules() {
+        return fareRules;
+    }
+
+    public List<FeedInfo> getFeedInfos() {
+        return feedInfos;
+    }
+
+    public List<Frequency> getFrequencies() {
+        return frequencies;
+    }
+
+    public List<Pathway> getPathways() {
+        return pathways;
+    }
+
+    public List<Route> getRoutes() {
+        return routes;
+    }
+
+    public List<ShapePoint> getShapePoints() {
+        return shapePoints;
+    }
+
+    public List<Stop> getStops() {
+        return stops;
+    }
+
+    public List<StopTime> getStopTimes() {
+        return stopTimes;
+    }
+
+    public List<Transfer> getTransfers() {
+        return transfers;
+    }
+
+    public List<Trip> getTrips() {
+        return trips;
+    }
+
+    public OtpTransitService build() {
+
+        createNoneExistingIds();
+
+        return new OtpTransitServiceImpl(agencies, calendarDates, calendars, fareAttributes, fareRules,
+                feedInfos, frequencies, pathways, routes, shapePoints, stops, stopTimes, transfers,
+                trips);
+    }
+
+    private void createNoneExistingIds() {
+        generateNoneExistingIds(feedInfos);
+    }
+
+    static <T extends IdentityBean<Integer>> void generateNoneExistingIds(List<T> entities) {
```

Comment:
I think several GTFS domain objects had an optional id, I removed the id from what I considered ValueObjects - making sure the ID was not used in OTP. The only one left is FeedInfo. I am not sure if it is the entire object or the id that is optional, but the OBA library generated these IDs and it might be that the only place where this break now are the tests witch do not use OBS to load data. It should be possible to get rid of this. I did not remove this because I do not know if this would break anyones GTFS import.